### PR TITLE
Add annotations to main files

### PR DESCRIPTION
**Pull request by KasperFranz**
This pull request adds annotations to some Python files which should make it easier to maintain :)

### CODE REVIEW

File: script.py [hunk 1]
```diff
@@ -144,13 +155,16 @@ def update_validator_database_and_reports():
                 continue
         process_given_area(cursor, entry)
         connection.commit()
-        #generate_webpage_with_error_output.write_index_and_merged_entries(cursor) # update after each run
+        # generate_webpage_with_error_output.write_index_and_merged_entries(cursor) # update after each run
+    commit_and_publish_changes_in_report_directory(
+        cursor)  # note that it is called not only here! But also at start of the function
     connection.close()
-    commit_and_publish_changes_in_report_directory() # note that it is called not only here! But also at start of the function
+
```
Comment (KasperFranz) [inline script.py:162]: I moved this connection.close() to after the commit_and_publish which caused an exception as well

**Comment by KasperFranz** [comment]
> I will think a bit about this, I have not used types in Python so far.
>
> Would it help you to use this code?

My main language is PHP, where I use types all the time, and I find it very helpful; I do not have a lot of experience with Python besides a few small hobby projects where I use types.

It makes it easier for editors to help with types e.g parameters for the execute function for cursor 

example with no types/annotation:
image


Example with types
image


If we have not used the correct type (e.g. sent in a database connection instead of a cursor) it will complain and it can get resolved

File: database.py [hunk 2]
```diff
@@ -34,34 +38,44 @@ def create_table_if_needed(cursor):
...
     returned = cursor.fetchall()
     if len(returned) == 0:
         return 0
     else:
         return returned[0][0]
 
-def get_bot_edit_timestamp(cursor, internal_region_name, bot_edit_type):
-    cursor.execute("SELECT bot_edit_timestamp FROM osm_bot_edit_log WHERE area_identifier = :area_identifier AND type = :type ORDER BY bot_edit_timestamp DESC LIMIT 1", {"area_identifier": internal_region_name, "type": bot_edit_type, })
+
+def get_bot_edit_timestamp(cursor: Cursor, internal_region_name: string, bot_edit_type: string):
```
Comment (matkoniecz) [inline database.py:62]: are enums available as types? `bot_edit_type` is more restrictive than just a string, is it possible to express it?

Comment (KasperFranz) [reply database.py:62]: Sure, we should be able to do that, is it correct the strings would be 
- handle_follow_wikipedia_redirect_where_target_matches_wikidata
- handle_wikidata_redirect
- add_wikipedia_tag_from_wikidata_tag
- add_wikidata_tag_from_wikipedia_tag

Comment (matkoniecz) [reply database.py:62]: wait, here it would be quite annoying as any new bot edit would require fiddling

**Comment by matkoniecz** [comment]
Sorry for things moving so slowly, but I moved to using this code in on my laptop for running code.

I will check is everything still running. If nothing will break I will merge it and will see how I feel with types in Python, on trial basis.

File: database.py (hunk 2 above)
Comment (KasperFranz) [reply database.py:62]: I agree :)

**Comment by KasperFranz** [comment]
> Sorry for things moving so slowly, but I moved to using this code in on my laptop for running code.
> 
> I will check is everything still running. If nothing will break I will merge it and will see how I feel with types in Python, on trial basis.

No worries,

I am running this on my local as well, I have not seen any issues so far

**Pull request merged**